extensions/vertx-core/deployment/src/main/java/io/quarkus/vertx/core/deployment/VertxCoreProcessor.java

```diff
@@ -69,12 +64,19 @@ IOThreadDetectorBuildItem ioThreadDetector(VertxCoreRecorder recorder) {
 
     @BuildStep
     @Record(value = ExecutionTime.RUNTIME_INIT)
-    CoreVertxBuildItem build(VertxCoreRecorder recorder, BeanContainerBuildItem beanContainer,
+    CoreVertxBuildItem build(VertxCoreRecorder recorder,
             LaunchModeBuildItem launchMode, ShutdownContextBuildItem shutdown, VertxConfiguration config,
+            BuildProducer<SyntheticBeanBuildItem> syntheticBeanBuildItemBuildProducer,
             BuildProducer<ServiceStartBuildItem> serviceStartBuildItem) {
 
-        Supplier<Vertx> vertx = recorder.configureVertx(beanContainer.getValue(), config,
+        Supplier<Vertx> vertx = recorder.configureVertx(config,
                 launchMode.getLaunchMode(), shutdown);
+        syntheticBeanBuildItemBuildProducer.produce(SyntheticBeanBuildItem.configure(Vertx.class)
+                .types(Vertx.class)
+                .scope(ApplicationScoped.class)
```

Comment:
@stuartwdouglas I need it to be a Singleton. Many clients are casting it to VertxInternal (I know it's bad). With ApplciaitonScoped it does not work anymore.

Reply:
https://github.com/quarkusio/quarkus/pull/8339